src/operations/booleans/trees/PolygonTreeNode.js

```diff
@@ -117,7 +112,7 @@ PolygonTreeNode.prototype = {
         nodes = queue[i]
         for (j = 0, l = nodes.length; j < l; j++) { // ok to cache length
           node = nodes[j]
-          if (node.children.length) {
+          if (node.children.length > 0) {
```

Comment:
carefull : that is not quite the same type of check
- if you check for 'length' you essentially check if it is an array
- if you check for length > 0 it means it is 100% always an array
Which applies in this case ?

Reply:
#2 As the change is to eliminate unnecessary push of children on line 116. The children is initially set to an empty set, so the check is really if node.children.length is true, which is always true unless length is 0. So, I corrected the expression.

Reply:
ok , missed the initial part , that is way good :)